### PR TITLE
Switch StakingRequestProvider to Redux

### PR DESCRIPTION
**Pull request by ericlowturing**
[Associated Notion Card](https://www.notion.so/dcf93eee667d4edeac4b2a927d211551?v=0fbbeae3210b49af8266e3495e7d8ba5&p=4b326190d408403cbcb2dacad096fe99)

-----

## 1. Pull request checklist

- [ ] Notion card has been moved to the Code Review column
- [ ] Notion card has a link to this PR
- [ ] A reviewer has been assigned to the Notion card

### CODE REVIEW

File: src/lib/hooks/useStaking.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
+import { useMemo, useCallback } from 'react';
```
Comment (domw30) [inline src/lib/hooks/useStaking.tsx:1]: Leave this for now as we can do this in it's own PR - We should extract the error text to a global error constant file and replace with all instances in the codebase, I think we re-use them quite a bit.

Comment (domw30) [inline src/lib/hooks/useStaking.tsx:1]: Leave this for now as we can do this in it's own PR - We should extract the error text in the file to a global error constant file and replace with all instances in the codebase, I think we re-use them quite a bit.

File: src/lib/hooks/useStaking.tsx [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+import {
+	createDomainMetadata,
+	UploadedDomainMetadata,
+	uploadToIPFS,
+} from 'lib/utils';
+import useNotification from 'lib/hooks/useNotification';
+import { useStakingController } from 'lib/hooks/useStakingController';
+import { useStakingRedux } from 'store/staking/hooks';
+
+export type UseStakingReturn = {
```
Comment (domw30) [inline src/lib/hooks/useStaking.tsx:19]: just a curious question mate...what is the difference using ```type``` over ```interface``` for the function props? I ask as I've been instructed in previous projects to use ```interface```

File: src/store/staking/actions.test.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	setStakingFulfillingRequest,
+	setStakingFulfilledRequest,
+} from './actions';
+import {
+	NFT_STATUS_CARD_1,
+	NFT_STATUS_CARD_2,
+	DOMAIN_REQUEST_AND_CONTENTS_1,
+	DOMAIN_REQUEST_AND_CONTENTS_2,
+} from './staking.mockData';
+
```
Comment (domw30) [inline src/store/staking/actions.test.ts:31]: Nice testing 🔥

File: src/store/staking/types.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+export type StakingState = {
+	requesting: NftStatusCard[];
+	requested: NftStatusCard[];
+	approving: DomainRequestAndContents[];
+	approved: DomainRequestAndContents[];
+	fulfilling: DomainRequestAndContents[];
+	fulfilled: DomainRequestAndContents[];
+};
+
+/**
```
Comment (domw30) [inline src/store/staking/types.ts:23]: I could be wrong but the comment and type names below look like they have slight typos? is ```stacking``` supposed to be ```staking```?

**Review by domw30 — APPROVED** [review]
Nice 👍  Approving with a couple of questions 😁

File: src/lib/hooks/useStaking.tsx (hunk 2 above)
Comment (ericlowturing) [reply src/lib/hooks/useStaking.tsx:19]: @domw30 You are right. I sometimes forget it :(

File: src/store/staking/types.ts (hunk 4 above)
Comment (ericlowturing) [reply src/store/staking/types.ts:23]: Good call - This is typo
